kedro/templates/project/{{ cookiecutter.repo_name }}/pyproject.toml

```diff
@@ -3,9 +3,16 @@ requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
 
 [project]
+requires-python = ">=3.9"
 name = "{{ cookiecutter.python_package }}"
 readme = "README.md"
-dynamic = ["dependencies", "version"]
+dynamic = ["version"]
+dependencies = [
+    "ipython>=8.10",
+    "jupyterlab>=3.0",
+    "notebook",
+    "kedro~={{ cookiecutter.kedro_version }}"
+]
 
 [project.scripts]
 {{ cookiecutter.repo_name }} = "{{ cookiecutter.python_package }}.__main__:main"
```

Comment:
Should we add the quotes here as well to be consistent with starters (https://github.com/kedro-org/kedro-starters/blob/main/spaceflights-pandas/%7B%7B%20cookiecutter.repo_name%20%7D%7D/pyproject.toml#L11)?